rotkehlchen/externalapis/coingecko.py

```diff
@@ -643,14 +626,23 @@ def query_historical_price(
 
         # no cache, query coingecko for daily price
         date = timestamp_to_date(timestamp, formatstr='%d-%m-%Y')
-        result = self._query(
-            module='coins',
-            subpath=f'{from_coingecko_id}/history',
-            options={
-                'date': date,
-                'localizatioen': 'false',
-            },
-        )
+        try:
+            result = self._query(
+                module='coins',
+                subpath=f'{from_coingecko_id}/history',
+                options={
+                    'date': date,
+                    'localizatioen': 'false',
+                },
+            )
+        except RemoteError as e:
+            rate_limited = e.error_code == HTTPStatus.TOO_MANY_REQUESTS
+            raise NoPriceForGivenTimestamp(
+                from_asset=from_asset,
+                to_asset=to_asset,
+                time=timestamp,
+                rate_limited=rate_limited,
```

Comment:
True, It is cached at https://github.com/rotki/rotki/pull/4933/files#diff-650521abff84664ac714d403c589c1d8c5b147bae64709b5df39804cbc20eabcR230

about why not in one line is just because for me it adds complexity reading the code but is fine